File: src/main/java/ai/ailia/llm/AiliaLLMMediaData.java
package ai.ailia.llm;

/**
 * Represents media data (image or audio) for multimodal LLM processing.
 * Corresponds to the AILIALLMMediaData C structure.
 */
public class AiliaLLMMediaData {
    /** Media type: "image" or "audio" (audio is reserved for future use) */
    public String mediaType;

    /** Path to the media file (UTF-8) */
    public String filePath;

    /** Raw media data (currently unsupported, reserved for future use) */
    public byte[] data;

    /** Size of the raw data */
    public int dataSize;

    /** Width for images (pixels), sample count for audio */
    public int width;

    /** Height for images (pixels), unused for audio (set to 0) */
    public int height;

    /**
     * Creates a new media data object for file-based input.
     *
     * @param mediaType The media type ("image" or "audio")
     * @param filePath The path to the media file
     * @param width Width for images (pixels), sample count for audio
     * @param height Height for images (pixels), 0 for audio
     */
    public AiliaLLMMediaData(String mediaType, String filePath, int width, int height) {
        this.mediaType = mediaType;
        this.filePath = filePath;
        this.data = null;
        this.dataSize = 0;
        this.width = width;
        this.height = height;
    }

    /**
     * Creates a new media data object for file-based input with default dimensions.
     * The dimensions will be determined automatically from the file.
     *
     * @param mediaType The media type ("image" or "audio")
     * @param filePath The path to the media file
     */
    public AiliaLLMMediaData(String mediaType, String filePath) {
        this(mediaType, filePath, 0, 0);
    }

    @Override
    public String toString() {
        return String.format("AiliaLLMMediaData[type=%s, path=%s, size=%dx%d]",
            mediaType, filePath, width, height);
    }
}

File: src/main/kotlin/ai/ailia/llm/AiliaLLMMediaData.kt
package ai.ailia.llm

/**
 * Represents media data (image or audio) for multimodal LLM processing.
 * Corresponds to the AILIALLMMediaData C structure.
 *
 * @property mediaType Media type: "image" or "audio" (audio is reserved for future use)
 * @property filePath Path to the media file (UTF-8)
 * @property data Raw media data (currently unsupported, reserved for future use)
 * @property dataSize Size of the raw data
 * @property width Width for images (pixels), sample count for audio
 * @property height Height for images (pixels), unused for audio (set to 0)
 */
data class AiliaLLMMediaData @JvmOverloads constructor(
    @JvmField val mediaType: String,
    @JvmField val filePath: String,
    @JvmField val width: Int = 0,
    @JvmField val height: Int = 0,
    @JvmField val data: ByteArray? = null,
    @JvmField val dataSize: Int = 0
) {
    override fun toString(): String =
        "AiliaLLMMediaData[type=$mediaType, path=$filePath, size=${width}x${height}]"

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as AiliaLLMMediaData

        if (mediaType != other.mediaType) return false
        if (filePath != other.filePath) return false
        if (width != other.width) return false
        if (height != other.height) return false
        if (data != null) {
            if (other.data == null) return false
            if (!data.contentEquals(other.data)) return false
        } else if (other.data != null) return false
        if (dataSize != other.dataSize) return false

        return true
    }

    override fun hashCode(): Int {
        var result = mediaType.hashCode()
        result = 31 * result + filePath.hashCode()
        result = 31 * result + width
        result = 31 * result + height
        result = 31 * result + (data?.contentHashCode() ?: 0)
        result = 31 * result + dataSize
        return result
    }
}
